src/include/platform/PersistedStorage.h

```diff
@@ -87,6 +91,139 @@ CHIP_ERROR Read(Key aKey, uint32_t & aValue);
  */
 CHIP_ERROR Write(Key aKey, uint32_t aValue);
 
+class KeyValueStore
+{
+public:
+    virtual ~KeyValueStore() = default;
+    /**
+     * @brief
+     *   Reads the value of an entry in the KVS. The value is read into the
+     *   provided buffer and the number of bytes read is returned. If desired,
+     *   the read can be started at an offset.
+     *
+     *   If the output buffer is too small for the value, Get returns
+     *   CHIP_ERROR_BUFFER_TOO_SMALL with the number of bytes read. The remainder of the
+     *   value can be read by calling get with an offset.
+     *
+     * @param[in]     key               The name of the key in which to get.
+     * @param[in,out] buffer            A buffer to read the value into.
```

Comment:
Why `void*` instead of `uint8_t*` if this is meant to be bags of bytes?

Reply:
void* was just to avoid needing to cast to uint8_t*, similar to memcpy.

Reply:
Yes, but it hides details.

For example, if the thing being stored/retrieved is a 16-bit int, is the idea that it's stored in native endianness and retrieved by just byte-copy?  The void* bit means the caller doesn't even think to ask this question, and maybe that's intended if the store is always interacted with from things that want the same endianness.....

Reply:
The KVS is intended to work on a single device and not be used as shared storage between multiple devices, so the endianness should be the same on the write and read, and the KVS will preserve the byte order/endianness.